### PR TITLE
Move to go mod + fix juju-db snap usage

### PR DESCRIPTION
**Pull request by hpidcock**
This PR moves juju/testing to go mod.

It also fixes using juju-db snap within unit tests.
This fix works with juju-db latest/candidate at the time of writing.

### CODE REVIEW

File: mgo.go [hunk 1]
```diff
@@ -173,10 +174,38 @@ func generatePEM(path string, serverCert *x509.Certificate, serverKey *rsa.Priva
...
+	var err error
+	mongopath, _, err := installedMongod.Get()
 	if err != nil {
 		return err
 	}
+
+	dbdir := ""
+
+	// Check for snap confined mongod.
+	if mongopath == "/snap/bin/juju-db.mongod" {
```
Comment (achilleasa) [inline mgo.go:186]: Is it possible to have both the snap and the apt version installed and attempting to use the apt version for tests (e.g. testing the 2.7 branch)? This change will always give precedence to the snap mongo. If that's the case perhaps we would need to have a version preference argument or check for a magic envvar

Comment (hpidcock) [reply mgo.go:186]: JUJU_MONGOD env var has preference over everything, so that could be the solution.

Comment (hpidcock) [reply mgo.go:186]: Also I don't think we'll be updating 2.7 anytime soon, or even updating the 2.7 import for this package. So safe to say we can go ahead with this.

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
!!build!!

**Comment by wallyworld** [comment]
!!build!!

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**